README.md

```diff
@@ -226,3 +237,104 @@ The only thing that should seem unfamiliar here is `__resourcePath` -- this is a
+
+## Scanning MDX Content
+
+### How It Works
+
+#### Sample `next.config.js` entry:
+
+```js
+// in next.config.js
+withMdxEnhanced({
+  scan: [
+    {
+      pattern: /<SomeComponent.*name=['"](.*)['"].*\/>/,
+      passToLayout: {
+        argumentName: 'someImportantKey',
+        value: matchData => matchData[1] // find the specific entry or simply pass a static value
+      }
+    }
+  ]
+})
+```
+
+If an MDX page uses `<SomeComponent />`
+
+```js
+---
+layout: 'docs-page'
+title: 'Advanced Docs'
+---
+
+import SomeComponent from '../../components/SomeComponent'
+
+This is some _really_ **advanced** docs content!
+
+<SomeComponent name="Find this text" />
+
+```
+
+This will produce an `Array` of matches.
+
+```js
+__scans: {
+  someImportantKey: ['match1', 'match2']
+}
+```
+
+It is then up to the developer to consume these values in the `Layout` with `__scans` that is passed in as a parameter. This is where some interesting use cases become possible!
+
+### `<script>` Use Case:
+
+Let's assume you want to support some external content feature such as a dynamic social media feed, photo or video embed, or some other interesting vendor-provided feature that would usually require
+
+1. A bit of HTML markup
+2. A vendor-provided script
+
+A good place for this script would likely be within the HTML document `<head>` or near the closing `</body>` - both of which will need to be handled by your `Layout` component if you're building an `MDX`-driven page.
+
+`MDX` Page
+
+```mdx
+---
+layout: 'docs-page'
+title: 'Intro Docs'
+---
+
+import EmbeddedContentFeed from '../../components/EmbeddedContentFeed'
+
+Here's some _introductory_ docs content
+
+<EmbeddedContentFeed /> // outputs <div data-id="foo" data-vendor-xyz-property="bar"></div>
+```
+
+`next.config.js`
+
+```js
+module.exports = withMdxEnhanced({
+  scan: {
+    hasEmbeddedContentFeed: {
+      pattern: /<EmbeddedContentFeed.*.*\/>/
+    }
+  }
+})()
+```
+
+The plugin returns `__scans : { hasEmbeddedContentFeed: ['/<EmbeddedContentFeed \/>/'] }`
+
+This can be used in the `Layout` Component to conditionally load the script
+
+```jsx
+export default plugIn => {
+  const __scans = plugIn.__scans
+  return function docsPageLayout({ children }) {
+    return (
+   <>
+      <Head>
+        {__scans.hasEmbeddedContentFeed && (
+          <script  src="//vendor.com/embed.js" ></script>
+        )}
+     </Head>
+//....
+```
```

Comment:
To be honest this example feels too specific to one use case to be part of the readme. I'd rather see a more thorough explanation of how `value` works in the section above and have this cut.

That being said, I'll defer to you & anyone else with opinions on whether it's valuable to make the final call -- I think my view might be a bit biased since I was involved in creating this feature, and for someone who was not it could well be a valuable piece of context for aiding understanding.

Reply:
:hocho: is good with me. I agree it is very specific